addons/html_editor/static/src/main/feff_plugin.js

```diff
@@ -31,6 +31,7 @@ export class FeffPlugin extends Plugin {
         normalize_handlers: this.updateFeffs.bind(this),
         clean_handlers: (root) => this.clean({ root, preserveSelection: true }),
         clean_for_save_handlers: this.clean.bind(this),
+        legit_zwnbsp_predicates: [(node, feffNodesToKeep) => feffNodesToKeep.has(node)],
```

Comment:
It's weird that the predicates would receive `feffNodesToKeep`. I suggest to simply move that check in the exclude function at call site. Just check if it's in the set or if one of the predicates return `true`.